python/mrtrix3/commands/dwimotioncorrect.py

```diff
@@ -0,0 +1,333 @@
+        if app.ARGS.priorweights:
+            import numpy as np
+            W0 = np.loadtxt('priorweights.txt')
+            W1 = np.loadtxt('sliceweights.txt')
+            np.savetxt('sliceweights.txt', W1 * W0)
+
+
+    def basisupdatestep(k, conf):
+        lboption = ' -lbreg {}'.format(conf['lbreg']) if conf['lbreg']>0 else ''
+        run.command('msshsvd recon-' + str(k) + '.mif -mask mask.mif -lmax ' + ','.join([str(l) for l in sorted(lmax)[::-1]]) + lboption + ' ' + ' '.join(rfs) + ' -force')
+
+
+    def rankreduxstep(k, conf):
+        lboption = ' -lbreg {}'.format(conf['lbreg']) if conf['lbreg']>0 else ''
+        fwhmoption = ' -fwhm {}'.format(round(vu*conf['reg-scale'], 2))
+        run.command('mrfilter recon-' + str(k) + '.mif smooth -' + fwhmoption + ' | ' +
+                    'msshsvd - -mask mask.mif -lmax ' + ','.join([str(l) for l in rlmax]) +
+                    lboption + ' ' + ' '.join(redrfs) + ' -proj rankredux.mif -force')
+
+
+    def registrationstep(k, conf):
+        if app.ARGS.fixedmotion:
+            return;
+        rcmd = 'dwislicealign {} rankredux.mif motion.txt -mask mask.mif -maxiter {}'.format(inputfn, conf['reg-iter'])
+        rcmd += ' -mb ' + (str(mb) if conf['svr'] else '0') + (' -init motion.txt' if k>0 else '')
+        rcmd += ssp_option + ' -force' + nthr
+        run.command(rcmd)
+        run.command('motionfilter motion.txt sliceweights.txt motion.txt -medfilt 5' + motfilt_option + ' -force')
+
+
+    def fieldalignstep(k):
+        global inputfn, voxweightsfn
+        if app.ARGS.fieldmap:
+            cmdopt = ' -motion motion.txt -fidx ' + ('0' if not app.ARGS.fieldidx else app.ARGS.fieldidx) + ' -force' if k > 0 else ''
+            run.command('mrfieldunwarp in.mif field.mif unwarped.mif' + cmdopt)
+            if app.ARGS.voxelweights:
+                run.command('mrfieldunwarp -nomodulation voxelweights.mif field.mif voxelweights-unwarped.mif' + cmdopt)
+            inputfn = 'unwarped.mif'
+            voxweightsfn = 'voxelweights-unwarped.mif'
+
+
+    #   __________ Motion correction __________
+
+    nepochs = len(config['epochs']) - 1
+    for k, epoch in enumerate(config['epochs']):
+        liveconfig = {**config['global']}
+        liveconfig.update(epoch)
+        # update template
+        fieldalignstep(k)
+        reconstep(k, liveconfig)
+        # end on last recon step
+        if k == nepochs:
+            break
+        # basis update and rank reduction
+        basisupdatestep(k, liveconfig)
+        rankreduxstep(k, liveconfig)
+        # slice weights
+        sliceweightstep(k)
+        # register template to volumes
+        registrationstep(k, liveconfig)
+
+
+    #   __________ Copy outputs __________
+
+    run.command(['mrconvert', f'recon-{nepochs}.mif', app.ARGS.output], mrconvert_keyval=f'recon-{nepochs}.mif', force=app.FORCE_OVERWRITE)
```

Comment:
```suggestion
    run.command(['mrconvert', f'recon-{nepochs}.mif', app.ARGS.output],
                mrconvert_keyval=app.ARGS.input,
                force=app.FORCE_OVERWRITE)
```

Purpose here is to have `command_history` in the output image reflect the invocation of `dwimotioncorrect`, rather than listing all of its internal steps.